docs/proposals/multi-arch-image-mgmt.md

```diff
@@ -0,0 +1,256 @@
+
+#### Hard to compose and update a multi-arch image
+
+To create a multi-arch image, ORAS users have to manually compose an OCI image index from one or several manifest JSON files, then use "oras manifest push" to push it to an OCI layout or a registry. Typical examples can be found from these two scripts:
+
+- ORAS user A: [POC: Create and install Postgres Trunk Binaries (github.com)](https://gist.github.com/theory/7dc164e5772cae652d838a1c508972ae#file-push_trunk-L22) and his blog post [POC: Distributing Trunk Binaries via OCI (justatheory.com)](https://justatheory.com/2024/06/trunk-oci-poc/)
+- ORAS user B: [Script to help create a simple mult-arch artifact (github.com)](https://gist.github.com/sajayantony/0c9d6436c03d531b1bbebe43249381cf)
+
+#### `docker buildx`/`docker` is unavailable
+
+Users are not able to install or use `docker buildx` or even no `docker` and its daemon service in some strict environments. Users need to seek for an alternative tool to create a mulit-arch image without relying on `docker` or `docker buildx`.
+
+## Proposed solution
+
+Ideally, if ORAS extends the ability to create and manage a multi-arch image from a local environment or a remote registry, the problems listed above can be resolved. Creating and managing a multi-arch image using image index format should be as easy as playing Legos. 
+
+The proposed CLI commands for managing a multi-arch image are listed below. The detailed use cases and subcommands are articulated in the CLI Spec section.
+
+- Create a multi-arch image: `oras manifest index create`
+- Update a multi-arch image: `oras manifest index update`
+- Inspect a multi-arch image: `oras manifest fetch`. Add two alias `oras manifest inspect` and `oras manifest show` for usability
+- Add annotations to a multi-arch image: `oras manifest create --annotation`
+
+The proposal creates a multi-arch image using an OCI image index in an OCI image layout as a local storage, then push the multi-arch image to the registry with ORAS. 
+
+### User scenario and desired experience
+
+Here is the sample workflow to create a multi-arch image using an image index locally and push it to the registry for deployment:
+
+![multi-arch image](./img/create-multi-arch.png)
+
+1. Assume there are two arch-specific images tagged as `v1-linux-amd64` and `v1-linux-arm64` in an OCI image layout called `layout-dir`. List the tags in the OCI image layout:
+
+```bash
+$ oras repo tags --oci-layout layout-dir 
+
+v1-linux-amd64 
+v1-linux-arm64
+v1-linux-arm/v7
+```
+
+2. Create a multi-arch image by combining two image manifests into an image index, tag it with `v1` and push the tagged image index to an OCI image layout `layout-dir` automatically. Add annotations to this image index at the same time:  
+
+```bash
+$ oras manifest index create --oci-layout layout-dir:v1 v1-linux-amd64 v1-linux-arm64 --annotation "platform=multi-arch" 
```

Comment:
A more meaningful annotation should be used in the spec.

Reply:
Updated annotations to align with OCI spec